### PR TITLE
Fix streaming YAML from STDIN in kubectl

### LINKED ISSUE #3448: `kubectl -f -` broken
```
Running: /home/dbsmith/code/k/src/github.com/GoogleCloudPlatform/kubernetes/cluster/../cluster/gce/../../cluster/../cluster/gce/../../_output/dockerized/bin/linux/amd64/kubectl create -f -
F0114 01:30:06.708409   12278 create.go:73] invalid character 'k' looking for beginning of value
```

If I stuff into a file and use that after -f, it works fine.

cc @smarterclayton @zmerlynn @satnam6502

### PR DESCRIPTION
**Pull request by smarterclayton**
Allows streaming of YAML documents from the CLI (if they are separated by \n---)
as per the YAML spec.  Should be moved into ghodss/yaml eventually.

Fixes #3448

### CODE REVIEW

File: pkg/util/yaml/decoder.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	}
+}
+
+// Decode unmarshals the next object from the underlying stream into the
+// provide object, or returns an error.
+func (d *YAMLOrJSONDecoder) Decode(into interface{}) error {
+	if d.decoder == nil {
+		buffer, isJSON := guessJSONStream(d.r, d.bufferSize)
+		if isJSON {
+			d.decoder = json.NewDecoder(buffer)
```
Comment (brendandburns) [inline pkg/util/yaml/decoder.go]: verbose glog your decision here?  For debugging purposes, in case you guess wrong?

Comment (smarterclayton) [inline pkg/util/yaml/decoder.go]: Yeah, V(4)?

**Comment by lavalamp** [comment]
LGTM- merging due to broken e2e. Thanks for quick fix.

File: pkg/kubectl/resource/builder_test.go [hunk 2]
```diff
@@ -25,6 +25,8 @@ import (
 	"reflect"
 	"testing"
 
+	"github.com/ghodss/yaml"
```
Comment (lavalamp) [inline pkg/kubectl/resource/builder_test.go]: nit: these usually go after the k8s packages

**Pull request merged**